client/src/featureform/register.py

```diff
@@ -4089,25 +4096,22 @@ def register_training_set(
             elif resource_label != ():
                 raise ValueError("A training set can only have one label")
 
-        if isinstance(label, LabelColumnResource):
-            label = label.name_variant()
```

Comment:
this is a good example to understand what's going on, by removing this we're not locked into the label's name variant until we finally go to create